datafusion/physical-plan/src/memory.rs

```diff
@@ -165,6 +167,7 @@ impl LazyMemoryExec {
     /// Create a new lazy memory execution plan
     pub fn try_new(
         schema: SchemaRef,
+        projection: Option<Vec<usize>>,
```

Comment:
this is technically a breaking API change -- it is minor but still breaking

What would you think about adding a new method like

```rust
impl LazyMemoryExec {
  pub fn with_projection(projection: Option<Vec<usize>>) -> Self { ... }
}
```
instead and leave the signature of `try_new` the same

That would avoid the API breakage (and likely reduce the size of this PR a bit as many callsites would not have to be changed)